iio/iio_axi_adc/iio_axi_adc.c

```diff
@@ -659,7 +448,101 @@ static ssize_t iio_axi_adc_read_dev(void *iio_inst, char *pbuf, size_t offset,
 }
 
 /**
- * @brief Registers a iio_axi_adc for reading/writing and parameterization of
+ * @brief Delete iio_device.
+ * @param iio_device - Structure describing a device, channels and attributes.
+ * @return SUCCESS in case of success or negative value otherwise.
+ */
+static ssize_t iio_axi_adc_delete_device_descriptor(
+	struct iio_device *iio_device)
+{
+	uint16_t i = 0;
+
+	if (!iio_device)
+		return FAILURE;
+
+	if (iio_device->channels) {
+		while (iio_device->channels[i]) {
+			if (iio_device->channels[i]->name)
+				free(iio_device->channels[i]->name);
+			if (iio_device->channels[i])
```

Comment:
The check is redundatnt. It is also checked in `while (iio_device->channels[i]) `